memcache/memcache_test.go

```diff
@@ -7,45 +7,52 @@ import (
 	"fmt"
 	"net"
 	"testing"
-
-	. "gopkg.in/check.v1"
 )
 
 const testServer = "localhost:11211"
 
-func Test(t *testing.T) { TestingT(t) }
-
-type S struct{}
-
-var _ = Suite(&S{})
-
-func (s *S) SetUpSuite(c *C) {
+func SetUpSuite() bool {
 	conn, err := net.Dial("tcp", testServer)
 	if err != nil {
 		// TODO: rather than skip the test, fall back to a faked memcached server
-		c.Skip(fmt.Sprintf("skipping test; no server running at %s", testServer))
+		fmt.Sprintf("skipping test; no server running at %s", testServer)
+		return false
 	}
 	conn.Write([]byte("flush_all\r\n")) // flush memcache
 	conn.Close()
+	return true
 }
 
-func (s *S) Test(c *C) {
+func TestMemCache(t *testing.T) {
+	if !SetUpSuite() {
+		t.SkipNow()
+	}
 	cache := New(testServer)
+	if cache == recover() {
```

Comment:
This is equivalent to `if cache == nil {`, and it's better to write that since it's simpler and more direct.

From the [Go spec](https://golang.org/ref/spec#Handling_panics):

> The return value of recover is nil if any of the following conditions holds:
>
> - panic's argument was nil;
> - the goroutine is not panicking;
> - recover was not called directly by a deferred function.

`recover` is not being called directly by a deferred function here, so its return value will always be `nil`. It can never be called when a panic is taking place, since it's not deferred, and it will never be executed if a panic happens before it runs.